ckanext/opendata_theme/processor.py

```diff
@@ -0,0 +1,397 @@
+from abc import abstractmethod
+from collections import defaultdict, OrderedDict
+import wcag_contrast_ratio as contrast
+
+__all__ = ['custom_style_processor']
+
+from ckanext.opendata_theme.color_conrast import get_contrast
+
+
+class AbstractParser:
+    color = None
+
+    @abstractmethod
+    def get_css_from_data(self, data):
+        return
+
+    def parse_form_data(self, data):
+        value = data.get(self.get_form_name())
+        self.color = value
+
+    @abstractmethod
+    def get_class_name(self):
+        raise NotImplemented
+
+    @abstractmethod
+    def get_form_name(self):
+        raise NotImplemented
+
+    @abstractmethod
+    def get_title(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_default_color(self):
+        raise NotImplementedError
+
+
+class AccountHeaderBackGroundColor(AbstractParser):
```

Comment:
```suggestion
class AccountHeaderBackGroundColor(AbstractParser):
    _class_name = '.account-masthead'
    _form_name = 'custom-css-account-header-background-color'
    _title = 'Account Header Background Color'
    _default_color = '#044187
    
    def get_css_from_data(self, data):
        self.parse_form_data(data)
        if self.color:
            return {'background': self.color.encode('utf-8')}
```

Reply:
I am wondering, it looks like a shorter version of the class, we lost some restriction of abstract classes, but they become more simple and readable.